### PR TITLE
Adds guide to installing Fairing, setting up GCP, and running the XGB…

### PR DESCRIPTION
**Pull request by joeliedtke**
Adds a guide to installing Fairing, setting up GCP, and running the XGBoost demo notebook.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/603)

### CODE REVIEW

**Comment by abhi-g** [comment]
relates to #595

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-603--competent-brattain-de2d6d.netlify.com/docs/fairing/gcp-local-notebook/

File: content/docs/fairing/gcp-local-notebook.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+*  Train an XGBoost model in a local notebook,
+*  Use Kubeflow Fairing to train an XGBoost model remotely on Kubeflow,
+*  Use Kubeflow Fairing to train an XGBoost model remotely on Google Cloud ML Engine, 
+*  Use Kubeflow Fairing to deploy a trained model to Kubeflow, and
+*  Call the deployed endpoint for predictions.
+
+This guide has been tested on Linux and Mac OS X. Currently, this guide has not been
+tested on Windows.
+
+## Clone the Kubeflow Fairing Repository
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: s/Repository/repository

(We use sentence case for headings within a doc.)

File: content/docs/fairing/gcp-local-notebook.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+weight = 30
++++
+
+This guide introduces you to using Kubeflow Fairing to train and deploy a
+model to Kubeflow on Google Kubernetes Engine, and Google Cloud ML Engine.
+As an example, this guide uses a local notebook to demonstrate how to:
+
+*  Train an XGBoost model in a local notebook,
+*  Use Kubeflow Fairing to train an XGBoost model remotely on Kubeflow,
+*  Use Kubeflow Fairing to train an XGBoost model remotely on Google Cloud ML Engine,
```
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/fairing/gcp-local-notebook.md]: There are a few more instances where we can remove "Google" on second usage. Please would you check for those.

File: content/docs/fairing/gcp-local-notebook.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
++++
+title = "Train and deploy on GCP from a local notebook"
+description = "Use Kubeflow Fairing to train and deploy a model on Google Cloud Platform from a local notebook."
+weight = 30
++++
+
+This guide introduces you to using Kubeflow Fairing to train and deploy a
+model to Kubeflow on Google Kubernetes Engine, and Google Cloud ML Engine.
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: Google Kubernetes Engine (GKE)

For later usages on this page, you can just use "GKE".

(GKE has been approved as a product name abbreviation, and is well known amongst our audience.)

File: content/docs/fairing/gcp-local-notebook.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+1.  Install the Python dependencies for the XGBoost demo notebook.
+
+    ```bash
+    pip3 install -r examples/prediction/requirements.txt
+    ```
+
+## Install and configure the Google Cloud SDK
+
+In order to use Kubeflow Fairing to train or deploy to Kubeflow on Google
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: GKE
Cloud ML Engine
GCP

File: content/docs/fairing/gcp-local-notebook.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    Cloud SDK][gcloud-install].
+
+1.  Use `gcloud` to set a default project.
+
+    ```bash
+    export PROJECT_ID=<your-project-id>
+    gcloud config set project $PROJECT_ID
+    ```
+
+1.  Kubeflow Fairing needs a service account to make API calls to Google Cloud
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: Just GCP is OK here, as we've already established the abbreviation.

File: content/docs/fairing/gcp-local-notebook.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+    Platform (GCP). The recommended way to provide Fairing with access to this
+    service account is to set the GOOGLE_APPLICATION_CREDENTIALS environmental
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: Use code formatting for env variable names.

File: content/docs/fairing/gcp-local-notebook.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    ls "$GOOGLE_APPLICATION_CREDENTIALS"
+    ```
+
+    The response should be something like this:
+
+    ```bash
+    /.../.../key.json
+    ```
+
+    If you do not have a service account, then let's create one and grant it
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: Remove "let's".

File: content/docs/fairing/gcp-local-notebook.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+    gcloud iam service-accounts keys create ~/key.json --iam-account $SA_NAME@$PROJECT_ID.iam.gserviceaccount.com
+    ```
+
+    Create the GOOGLE_APPLICATION_CREDENTIALS environmental variable.
+
+    ```bash
+    export GOOGLE_APPLICATION_CREDENTIALS=~/key.json
+    ```
+
+## Docker setup
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: s/Docker setup/Set up Docker

(To make the headings parallel.)

File: content/docs/fairing/gcp-local-notebook.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+You need to have Docker installed to use Kubeflow Fairing. Fairing packages
+your code as a docker image and executes it in the remote cluster. To check
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: s/docker/Docker

File: content/docs/fairing/gcp-local-notebook.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+   about how this affects security in your system, see the guide to the
+   [Docker daemon attack surface][docker-attack].
+
+Authorize Docker to access your GCP Container Registry. 
+
+```bash
+gcloud auth configure-docker
+```
+
+# Setting up Kubeflow
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: This should be a level 2 heading.

Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: s/Setting up/Set up

File: content/docs/fairing/gcp-local-notebook.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+
+Use the following instructions to set up and configure your Kubeflow and
+development environments for training and prediction from Kubeflow Fairing.
+
+1.  You need to have kubectl installed to use Kubeflow Fairing. To set up
+    kubectl, follow the instructions in the guide to [installing and setting
+    up kubectl][kubectl-install].
+
+1.  If you do not have a Kubeflow environment, follow the guide to [deploying
+    Kubeflow on GKE][kubeflow-install] to set up your Kubeflow environment.
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: This link has changed recently:
https://www.kubeflow.org/docs/gke/deploy/

File: content/docs/fairing/gcp-local-notebook.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+    The guide provides two options for setting up your environment:
+
+    *  The [Kubeflow deployment user interface][kubeflow-deploy] is an easy
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: This link is broken - it needs a "d" at the end.

File: content/docs/fairing/gcp-local-notebook.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+       way for you to set up a Google Kubernetes Engine cluster with Kubeflow
+       installed, or
+    *  You can deploy Kubeflow using the [command line][kubeflow-install].
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: This link is now:
https://www.kubeflow.org/docs/gke/deploy/deploy-cli/

File: content/docs/fairing/gcp-local-notebook.md [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+   Docker](https://docs.docker.com/install/).
+*  If you get an error like `Error response from daemon: Bad response from
+   Docker engine`, then [restart your docker daemon][docker-start].
+*  If you are using Linux and you use sudo to access Docker, follow these
+   steps to [add your user to the docker group][docker-non-root]. Note, the
+   docker group grants privileges equivalent to the root user. To learn more
+   about how this affects security in your system, see the guide to the
+   [Docker daemon attack surface][docker-attack].
+
+Authorize Docker to access your GCP Container Registry.
```
Comment (sarahmaddox) [inline content/docs/fairing/gcp-local-notebook.md]: This is the first time we mention Container Registry on this page, and it's not mentioned in the Kubeflow GCP setup guide. It's probably worth adding a link:
https://cloud.google.com/container-registry/

**Comment by sarahmaddox** [comment]
Relates to issue https://github.com/kubeflow/website/issues/593

**Comment by sarahmaddox** [comment]
Relates to https://github.com/kubeflow/website/issues/593

File: content/docs/fairing/gcp-local-notebook.md [hunk 15]
```diff
@@ -0,0 +1,228 @@
++++
+title = "Train and deploy on GCP from a local notebook"
```
[the remark this answers is not in the document]
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Thanks for pointing that out, changed.

File: content/docs/fairing/gcp-local-notebook.md [hunk 16]
```diff
@@ -0,0 +1,228 @@
...
+description = "Use Kubeflow Fairing to train and deploy a model on Google Cloud Platform from a local notebook."
```
[the remark this answers is not in the document]
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Good point, changed.

File: content/docs/fairing/gcp-local-notebook.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Ack, I should have gone back through this doc to check for this, sorry for the initial state of this doc. This has been updated.

File: content/docs/fairing/gcp-local-notebook.md (hunk 1 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Thanks for pointing that out, changed.

File: content/docs/fairing/gcp-local-notebook.md (hunk 3 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 4 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 5 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 6 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Thanks for pointing that out. I also updated the docker section to use code formatting when referring to the docker security group. Please let me know if that should be formatted differently.

File: content/docs/fairing/gcp-local-notebook.md (hunk 7 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 8 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: done

File: content/docs/fairing/gcp-local-notebook.md (hunk 9 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 10 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Ack, done

Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 11 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Thanks for pointing that out. Done.

File: content/docs/fairing/gcp-local-notebook.md (hunk 12 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Ack! Done.

File: content/docs/fairing/gcp-local-notebook.md (hunk 13 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done.

File: content/docs/fairing/gcp-local-notebook.md (hunk 14 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

**Comment by sarahmaddox** [comment]
Thanks! This LGTM. I'll add a hold to give @karthikv2k the opportunity to review.

/lgtm
/approve
/hold

File: content/docs/fairing/gcp-local-notebook.md [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+1.  Use `gcloud` to set a default project.
+
+    ```bash
+    export PROJECT_ID=<your-project-id>
+    gcloud config set project $PROJECT_ID
+    ```
+
+1.  Kubeflow Fairing needs a service account to make API calls to GCP. The
+    recommended way to provide Fairing with access to this
+    service account is to set the `GOOGLE_APPLICATION_CREDENTIALS` environmental
```
Comment (karthikv2k) [inline content/docs/fairing/gcp-local-notebook.md]: "environmental variable" - "environment variable"?

File: content/docs/fairing/gcp-local-notebook.md [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+    /.../.../key.json
+    ```
+
+    If you do not have a service account, then create one and grant it
+    access to the required roles.
+
+    ```bash
+    export SA_NAME=<your-sa-name>
+    gcloud iam service-accounts create $SA_NAME
+    gcloud projects add-iam-policy-binding $PROJECT_ID --member serviceAccount:$SA_NAME@$PROJECT_ID.iam.gserviceaccount.com  --role 'roles/editor'
```
Comment (karthikv2k) [inline content/docs/fairing/gcp-local-notebook.md]: The rendering of this code block is not ideal. The long command is putting a horizontal scroll that hides the last command. Probably split the long line using "\"?

File: content/docs/fairing/gcp-local-notebook.md [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+    ```
+
+    Create a key for your service account.
+
+    ```bash
+    gcloud iam service-accounts keys create ~/key.json --iam-account $SA_NAME@$PROJECT_ID.iam.gserviceaccount.com
```
Comment (karthikv2k) [inline content/docs/fairing/gcp-local-notebook.md]: same here.

File: content/docs/fairing/gcp-local-notebook.md [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+    ```
+
+    Create the `GOOGLE_APPLICATION_CREDENTIALS` environmental variable.
```
Comment (karthikv2k) [inline content/docs/fairing/gcp-local-notebook.md]: environmental -> environment?

File: content/docs/fairing/gcp-local-notebook.md [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+```bash
+gcloud auth configure-docker
+```
+
+## Set up Kubeflow
+
+Use the following instructions to set up and configure your Kubeflow and
+development environments for training and prediction from Kubeflow Fairing.
+
+1.  You need to have kubectl installed to use Kubeflow Fairing. To set up
```
Comment (karthikv2k) [inline content/docs/fairing/gcp-local-notebook.md]: >You need to have kubectl installed to use Kubeflow Fairing.

We don't for GCP installations. We use gcloud to get the k8s creds.

File: content/docs/fairing/gcp-local-notebook.md (hunk 17 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Thank you for pointing this out, for whatever reason I always thought that they were environmental variables and not environment variables. Good to know.

File: content/docs/fairing/gcp-local-notebook.md (hunk 20 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 21 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done.

File: content/docs/fairing/gcp-local-notebook.md (hunk 19 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Done

File: content/docs/fairing/gcp-local-notebook.md (hunk 18 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: Did you mean to split the long line with a \ or ""? I split the line with a \, let me know if you would like it split differently.

**Comment by karthikv2k** [comment]
/lgtm
Thanks @joeliedtke for the changes.

File: content/docs/fairing/gcp-local-notebook.md (hunk 21 above)
Comment (joeliedtke) [reply content/docs/fairing/gcp-local-notebook.md]: I also revised the wording on the "update kubeconfig" step to remove the reference to kubectl.

**Comment by sarahmaddox** [comment]
Fixes https://github.com/kubeflow/website/issues/361

/hold cancel

**Comment by sarahmaddox** [comment]
Fixes https://github.com/kubeflow/website/issues/566

**Pull request merged**